hypertrace-core-graphql-trace-schema/src/main/java/org/hypertrace/core/graphql/trace/dao/GatewayServiceTraceDao.java

```diff
@@ -46,7 +58,20 @@
 
   @Override
   public Single<TraceResultSet> getTraces(TraceRequest request) {
-    return this.requestBuilder
+    try {
+      return tenantToTraceFetchTime.computeIfAbsent(
+          request.context().getTenantId().orElse("NOT_AVAILABLE"),
+          tenantId ->
+            PlatformMetricsRegistry.registerTimer(TRACE_FETCH_LATENCY, Map.of("tenantid", tenantId)))
+          .recordCallable(getTracesCallable(request));
+    } catch (Exception e) {
+      LOG.error("Failed to get traces for tenant {}", request.context().getTenantId());
+    }
+    return null;
+  }
+
+  private Callable<Single<TraceResultSet>> getTracesCallable(TraceRequest request) {
```

Comment:
This timer assumes the callable is blocking, but it's returning an async single - that's the timing we want to measure. What kind of async functionality do we have?

Reply:
@samarth-gupta-traceable  We should look at Micrometer to see what kind of async support exists for metric tracking.

Reply:
@buchi-busireddy  @aaron-steinfeld   micrometer has `@Timed` annotation that works for async methods as well. See here: https://micrometer.io/docs/concepts#_the_timed_annotation
however I am not sure how will this integrate with the metric framework wrapper and registry that we have written.

Reply:
That doesn't look promising to me - I suspect we'll need to make our own wrapper for rxjava.